content/docs/rfcs/64/README.md

```diff
@@ -0,0 +1,258 @@
+- The optional `version` attribute SHOULD be set to `0`. It MUST be interpreted as `0` if not present.
+- The mandatory `timestamp` attribute MUST contain the Unix epoch time at which the message was generated by the application. The value MUST be in nanoseconds. It MAY contain a fudge factor of up to 1 seconds in either direction to improve resistance to timing attacks.
+- The optional `ephemeral` attribute MUST be set to `true` if the message should not be persisted by the Waku Network.
+- The optional `rate_limit_proof` attribute SHOULD be populated with the RLN proof as set out in dhjfhjhd. Messages with this field unpopulated MAY be discarded from the network by relayers.
+
+### Message Size
+
+Any Waku Message published to the network MUST NOT exceed an absolute maximum size of `150` kilobytes, attributes included.
+The average size of Waku Messages publised by each publisher SHOULD NOT exceed `3` kilobytes.
+
+### Message Validation
+
+Relay nodes MUST apply [gossipsub v1.1 validation](https://github.com/libp2p/specs/blob/c96c9ec5909d64fe020d7630f3fd982bc18fd06a/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators) to each relayed message and
+SHOULD apply all of the rules set out in the section below to determine the validity of a message.
+Validation has one of three outcomes,
+repeated here from the [gossipsub specification](https://github.com/libp2p/specs/blob/c96c9ec5909d64fe020d7630f3fd982bc18fd06a/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators) for ease of reference:
+1. Accept - the message is considered valid and it MUST be delivered and forwarded to the network
+2. Reject - the message is considered invalid, MUST be rejected and SHOULD trigger a gossipsub scoring penalty against the transmitting peer
+3. Ignore - the message SHOULD NOT be delivered and forwarded to the network, but this MUST NOT trigger a gossipsub scoring penalty against the transmitting peer.
+
+The following validation rules are defined:
+
+#### Decoding failure
+
+If a message fails to decode as a valid [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/).
+the relay node MUST _reject_ the message.
+This SHOULD trigger a penalty against the transmitting peer.
+
+#### Invalid timestamp
+
+If a message has a timestamp deviating by more than `20` seconds
+either into the past or the future
+when compared to the relay node's internal clock,
+the relay node MUST _ignore_ the message.
+
+#### Free bandwidth exceeded
+
+If a message contains no RLN proof
+and the current bandwidth utilization on the shard the message was published to
+equals or exceeds `1` Mbps,
+the relay node SHOULD _ignore_ the message.
+
+#### Invalid RLN epoch
+
+If a message contains an RLN proof
+and the `epoch` attached to the proof deviates by more than `20` seconds
+from the relay node's own `epoch`,
+the relay node MUST _ignore_ the message.
```

Comment:
Same comment as above, This may lead to DOS'ing, should we define a rate-limit for ignored messages which should eventually lead to peer disconnection if it continues?

Reply:
We do reject messages with an invalid epoch :)

Reply:
Right, I agree here that a stronger _reject_ may be necessary for RLN.

Reply:
Addressed in [74ba3ef](https://github.com/vacp2p/rfc/pull/616/commits/74ba3efb1dbc338ecccbc40276db2ba561dff68f)